packages/NodeCollector/BinaryOpTreeRootLocator.php

```diff
@@ -20,20 +24,32 @@ final class BinaryOpTreeRootLocator
      */
     public function findOperationRoot(Expr $expr, string $binaryOpClass): Expr
     {
-        /** @var ?Expr $parentNode */
         $parentNode = $expr->getAttribute(AttributeKey::PARENT_NODE);
-        if ($parentNode === null || $parentNode::class !== $binaryOpClass) {
+        if (! $parentNode instanceof Node) {
             return $expr;
         }
 
-        assert($parentNode instanceof BinaryOp);
-        $isLeftChild = $parentNode->left === $expr;
-        $isRightChild = $parentNode->right === $expr;
-        $isRightChildAndNotBinaryOp = $isRightChild && $expr::class !== $binaryOpClass;
-        if ($isLeftChild || $isRightChildAndNotBinaryOp) {
+        if ($parentNode::class !== $binaryOpClass) {
+            return $expr;
+        }
+
+        if (! $parentNode instanceof BinaryOp) {
+            return $expr;
+        }
+
+        if ($parentNode->left === $expr) {
             return $this->findOperationRoot($parentNode, $binaryOpClass);
         }
 
-        return $expr;
+        $isRightChild = $parentNode->right === $expr;
+        if (! $isRightChild) {
```

Comment:
Also it is weird that it moved the `$isLeftChild` condition into `if` but not `$isRightChild`.